cf/canaries.yaml

```diff
@@ -155,14 +180,68 @@ Resources:
                          !Sub "https://ui-default.scp-${Environment}.biomage.net/"]
       SuccessRetentionPeriod: 2
       RuntimeVersion: syn-nodejs-puppeteer-3.3
-      Schedule: 
+      Schedule:
         Expression: rate(5 minutes)
       StartCanaryAfterCreation: True
 
-  CanarySNSTopic: 
+  CanarySNSTopicCMK:
+    Type: AWS::KMS::Key
+    Properties:
+      Description: Customer Managed Key (CMK) to sign SNS topic at rest
+      Enabled: True
+      EnableKeyRotation: True
+      PendingWindowInDays: 7
+      KeyPolicy:
+        Version: "2012-10-17"
+        Id: canary-sns-topic-key
+        Statement:
+          - Sid: Enable IAM User Permissions
+            Effect: Allow
+            Principal:
+              AWS: !Sub "arn:aws:iam::${AWS::AccountId}:root"
+            Action: 'kms:*'
+            Resource: '*'
+          - Sid: Allow users to manage key
+            Effect: Allow
+            Principal:
+              AWS:
+              - !Sub "arn:aws:iam::${AWS::AccountId}:user/ibabukova"
+              - !Sub "arn:aws:iam::${AWS::AccountId}:user/polav"
```

Comment:
These permissions enable the user to delete the key or the CF stack which includes the key.

Unfortunately we can not give these permissions to a group, only users or roles. Meanwhile, our main avenue of giving permission is using the group `engineering-{Environment}-access`. So I give the permission to manage it to Pol and Iva.

Reply:
I think we already have permissions as admins for everything. I'm not sure if we need to add any permissions at all. This key should not be updated manually ever right? only managed by cloudformation. If that's the case, I'd remove these permissions

Reply:
We can create CF stacks using the IAC actions, but to delete stacks, right now we have to delete it manually (or write a new IAC workflow to delete the stack). Adding this permission enables the users listed to delete the CF stack containing this key manually, in case of failure stack creation failure. Without these permissions, only the root account can delete the stack. I think even the IAC can not delete this key, as it is using the user `ci-user-iac`, not root.

Reply:
I'd be surprised if the admins can't removed that. I would not add these permissions as we are not 100% sure we can't do it already nor we know that we'll need to remove them manually. We can always add it later if we need. I'd err in the side of less code always. Also this won't work with multiple deployments which might have different users.

Reply:
removed 👍